okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -356,10 +351,20 @@ public void run() {
       try {
         // Read until the underlying socket closes.
         while (frameReader.nextFrame(this)) {
+          // No need synchronization here, since frameReader is not closed
+          // yet, it's fine to read one more frame if the race happen.
+          if (stopped) {
+            break;
+          }
         }
       } catch (IOException ioe) {
         abort(ioe);
       } finally {
+        try {
+          frameReader.close();
+        } catch (IOException ignored) {
```

Comment:
If there is an I/O error, it's more likely caught and handled already, even when everything else goes well, but we happen to encounter an error here, we may not care much about it since we are done with this transport.